src/app/core/services/game/discord.ts

```diff
@@ -0,0 +1,56 @@
+export interface DiscordRPCStatus {
+  isInGame: boolean;
+  isMakingCharacter: boolean;
+  background: string;
+  playerName: string;
+
+  state: string;
+  details: string;
+}
+
+const status: DiscordRPCStatus = {
+  isInGame: false,
+  isMakingCharacter: false,
+  background: '',
+  playerName: '',
+
+  state: 'In a menu',
+  details: ''
+};
+
+export const getDiscordRPCStatus = (): DiscordRPCStatus => status;
+
+export const updateDiscordRPCStatus = (): void => {
+  status.state = status.isInGame ? 'Wandering the world' :
+    status.isMakingCharacter ? 'Planning their next move' : 'In a menu';
+  status.details = status.isInGame ? `Playing as ${status.playerName}, ${status.background}` : '';
+
+  (window as any).discordRPCStatus = status;
+};
+
+export const setDiscordRPCStatus =
+  (isInGame: boolean, isMakingCharacter: boolean,
```

Comment:
this signature should instead be an object instead of a list of params; one or two params is fine, but when you get too many it should really be an object instead. you can bind that object to an interface if needed as well